client/components/profilePage/allUsersModal.jsx

```diff
@@ -5,8 +5,12 @@ import axios from 'axios';
 export default function AllUsers({
   handleClose,
   show,
+  user,
 }) {
 
+  const addFriend = (newFriendId) => {
+    axios.post('/user/friend', {userId: user._id, friendId: newFriendId})
+  }
   const [allUsers, setAllUsers] = useState([]);
```

Comment:
For future reference, let's keep state variable declaration at the top of function if we can, so we don't run into variable assignment errors by using state variables before the useState hook